app/static/tests/unit/store/run/mutations.test.ts

```diff
@@ -202,6 +202,22 @@ describe("Run mutations", () => {
         expect(state.parameterSetResults).toStrictEqual({ Set2: { value: "test 2" } });
     });
 
+    it("swaps parameter sets with current parameter values", () => {
```

Comment:
Should probably also test that state is unchanged when `if` condition is not met

Reply:
I have included tests for if statements failing and making sure the state is the same